src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,125 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.InvalidUserDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static StorageDao storageDao;
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        user = new User();
+    }
+
+    @Test
+    void register_userLoginOnNull_notOk() {
+        user.setAge(23);
+        user.setPassword("qwer1234");
+        Exception exception = assertThrows(InvalidUserDataException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login can't be empty!", exception.getMessage());
+    }
+
+    @Test
+    void register_userLoginIsEmpty_notOk() {
+        user.setLogin("");
+        user.setAge(23);
+        user.setPassword("qwer1234");
+        Exception exception = assertThrows(InvalidUserDataException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login can't be empty!", exception.getMessage());
+    }
+
+    @Test
+    void register_userLoginHasTheSameLogin_notOk() {
+        user.setLogin("dima");
+        user.setAge(23);
+        user.setPassword("qwer1234");
+        storageDao.add(user);
+        User user1 = new User();
+        user1.setLogin("dima");
+        user1.setAge(32);
+        user1.setPassword("qws4123123");
+        Exception exception = assertThrows(InvalidUserDataException.class,
+                () -> registrationService.register(user1));
+        assertEquals("User with this login "
+                + user1.getLogin() + " is already registered!", exception.getMessage());
```

Comment:
We dont need to to it so hard? You can do it adding initialized user to list and try to do in again in "assertThrows"